query-engine/core/src/interpreter/query_interpreters/inmemory_record_processor.rs

```diff
@@ -188,3 +188,54 @@ impl InMemoryRecordProcessor {
         self.take.or(self.skip).is_some() || self.cursor.is_some()
     }
 }
+
+pub struct InMemoryRecordProcessorBuilder {
+    args: QueryArguments,
+}
+
+impl InMemoryRecordProcessorBuilder {
+    pub fn new(model: Model, order_by: Vec<OrderBy>, ignore_skip: bool, ignore_take: bool) -> Self {
+        Self {
+            args: QueryArguments {
+                model,
+                cursor: None,
+                take: None,
+                skip: None,
+                filter: None,
+                order_by,
+                distinct: None,
+                ignore_skip,
+                ignore_take,
+            },
+        }
+    }
+    pub fn _cursor(mut self, cursor: SelectionResult) -> Self {
+        self.args.cursor = Some(cursor);
+        self
+    }
+
+    pub fn _take(mut self, take: i64) -> Self {
+        self.args.take = Some(take);
+        self
+    }
+
+    pub fn _skip(mut self, skip: i64) -> Self {
+        self.args.skip = Some(skip);
+        self
+    }
+
+    pub fn _filter(mut self, filter: Filter) -> Self {
+        self.args.filter = Some(filter);
+        self
+    }
+
+    pub fn distinct(mut self, distinct: &mut Option<FieldSelection>) -> Self {
+        self.args.distinct = distinct.clone();
+        *distinct = None;
+        self
+    }
```

Comment:
Am I missing something or are these functions never used? Also, why are they prefixed with an underscore?

Reply:
Yep, that's why they're pre-fixed with an underscore. We can leave them out if we'd prefer, I just thought it'd be prudent to have them there for the future where we start doing these in-db too 🤔